src/Extension/FluentExtension.php

```diff
@@ -679,6 +683,91 @@ public function onAfterDuplicate($original, $doWrite, $relations): void
                 SQLUpdate::create($localisedTable, $fieldsToUpdate, ['RecordID' => $toID])->execute();
             }
         }
+
+        // We need to handle duplicating relations in `localised_copy` into additional locales
+
+        // We don't have any localised relations to cover
+        if (count($copyHasOneRelations) === 0) {
+            return;
+        }
+
+        $currentLocale = FluentState::singleton()->getLocale();
+
+        // We don't have a current locale which indicates that the Fluent setup is incomplete - bail out
+        if (!$currentLocale) {
+            return;
+        }
+
+        $locales = $this->getLocaleCodesForModel($owner);
+
+        // Current locale can be skipped as it was already handled correctly
+        $locales = array_diff($locales, [$currentLocale]);
+
+        // No locales need to be actioned
+        if (count($locales) === 0) {
+            return;
+        }
+
+        $ownerVersioned = $owner->hasExtension(Versioned::class);
+
+        foreach ($locales as $locale) {
+            FluentState::singleton()->withState(
+                static function (FluentState $state) use ($owner, $original, $ownerVersioned, $locale, $copyRelations): void {
+                    $state->setLocale($locale);
+
+                    $localisedOwner = DataObject::get($owner->ClassName)->byID($owner->ID);
+                    $localisedOriginal = DataObject::get($original->ClassName)->byID($original->ID);
```

Comment:
What's the difference between these, and why do we need both? I assume this is what you meant by "fixed data source for duplicated models" - we might need some inline comments here so it's clear what these data sources represent and why we need both of them.

Reply:
Added some more comments, here is how this looks like in a local test:

* original page (Int locale)
* -> original text block (Int locale)
* original page (Au locale)
* -> original text block (Au locale)

Trigger duplicate on original page in Int locale.

We get this:

* duplicated page (Int locale)
* -> duplicated text block (Int locale)
* duplicated page (Au locale)
* -> duplicated text block (Au locale)

Content mapping:

* original page (Int locale) - duplicated page (Int locale)
* original text block (Int locale) - duplicated text block (Int locale)
* original page (Au locale) - duplicated page (Au locale)
* original text block (Au locale) - duplicated text block (Au locale)